src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -189,7 +199,7 @@ private void PrepareServices()
         void addServiceAppResource(Service svc, IResource producingResource, ServiceBindingAnnotation sba)
         {
             svc.Spec.Protocol = PortProtocol.FromProtocolType(sba.Protocol);
-            svc.Spec.AddressAllocationMode = AddressAllocationModes.IPv4Loopback;
+            svc.Spec.AddressAllocationMode = AddressAllocationModes.Localhost;
```

Comment:
Because `AddressAllocationMode` was not a `{ get; set; }`, the allocation mode was never actually being set to `IPv4Loopback`--resulting in the default behavior, `Localhost`. In order to not change behavior, I'm changing this line to `Localhost`. We can change it back to `IPv4Loopback` if we want to.